src/main.rs

```diff
@@ -797,6 +800,17 @@ fn set_henvcfg(cpu_info: &CpuInfo) {
     }
 }
 
+// Sets the hstateen CSR.
+fn set_hstateen(cpu_info: &CpuInfo) {
+    // If sstateen is present, enable supported extensions.
+    if cpu_info.has_sstateen() {
+        CSR.hstateen0.modify(hstateen0::ctr.val(1));
+        CSR.hstateen0.modify(hstateen0::aia_imsic.val(1));
+        CSR.hstateen0.modify(hstateen0::aia.val(1));
+        CSR.hstateen0.modify(hstateen0::csrind.val(1));
+    }
```

Comment:
Add the envcfg bit as well ? Linux kernel with pointer masking and CFI will touch senvcfg.

Reply:
done, thanks!